src/critnib/critnib.c

```diff
@@ -653,7 +638,9 @@ static struct critnib_leaf *find_successor(struct critnib_node *__restrict n) {
     while (1) {
         unsigned nib;
         for (nib = 0; nib <= NIB; nib++) {
-            if (n->child[nib]) {
+            struct critnib_node *m;
+            utils_atomic_load_acquire_ptr((void **)&n->child[nib], (void **)&m);
+            if (m) {
                 break;
             }
         }
```

Comment:
What is a rationale for this change. Was old code wrong / not safe?

Reply:
This is it: https://github.com/oneapi-src/unified-memory-framework/actions/runs/13587677512/job/37987102380

Reply:
@ldorau sanitizer might report false positives. Are we sure it is a real issue?

Reply:
It was needed only by #1143. After adding a lock to #1143 it is not required any more.

Reply:
Which lock? What about scalability?

Reply:
It is still needed now: https://github.com/oneapi-src/unified-memory-framework/actions/runs/13650462662/job/38158581307

Reply:
Can anyone explain rationale why this change is needed? I need explanation why from logical point of view, of this code. That it suppress/fixes an sanitizer bug do not explain anything. From my perspective this change is like doing random changes until sanitizer is fine.

Reply:
removed
@ldorau please add this change to @1143 as it is needed (or not?) there

Reply:
> removed @ldorau please add this change to #1143 as it is needed (or not?) there

OK